context/app/static/js/shared-styles/snackbars/store.ts

```diff
@@ -52,4 +52,14 @@ const createStore = () =>
     toastWarning: (message, key) => get().openSnackbar(message, 'warning', key),
   }));
 
-export { Provider as SnackbarProvider, useStore as useSnackbarStore, createStore };
+const useSnackbarActions = () => {
+  return useStore((store) => ({
+    toastError: store.toastError,
+    toastInfo: store.toastInfo,
+    toastWarning: store.toastWarning,
+    toastSuccess: store.toastSuccess,
+    closeSnackbar: store.closeSnackbar,
+  }));
+};
```

Comment:
I lean towards not selecting the entire state by default as each component will update when any of the variables change even if the component does not depend on them. We could implement something like https://docs.pmnd.rs/zustand/guides/auto-generating-selectors if writing selectors ourselves is too much boilerplate.

Reply:
Thanks for the heads up! I agree completely about not selecting the entire state (since that would be more or less identical to using context). The auto generating selectors functionality seems very useful, and would be a solid improvement for us to make once we're on zustand v4.

**TL;DR:** - I can take this helper out and replace it with specific selectors where needed, but I'm not sure how much benefit that would bring in this case since these are just functions.

To my understanding, since the keys accessed/returned by this helper are functions, they shouldn't be updating at all; the idea was that this helper would let us skip making a separate selector for each case where we just need to be able to send a snackbar message without necessarily needing to read it. Additionally, if zustand functions _are_ regenerated on each state update, wouldn't they all be regenerated regardless of whether they're accessed separately or all together in a helper like this?